src/test/neorados/test_neorados.cc

```diff
@@ -396,42 +425,39 @@ TEST_F(TestNeoRADOSCoro, MakeWithBuilder) {
     }
   );
 
-  ASSERT_THROW(
+  EXPECT_THROW(
     ctx.run(),
     boost::system::system_error);
 }
 
 TEST_F(TestNeoRADOSCoro, CreatePool) {
   boost::asio::io_context ctx;
   boost::system::error_code ec;
-  unsigned pool = 0;
+  const std::string pool{"piscine"};
+  unsigned pool_id = 0;
 
   boost::asio::co_spawn(ctx, 
     [&]() -> boost::asio::awaitable<void> {
       auto rados = co_await RADOS::Builder{}.build(ctx, boost::asio::use_awaitable);
-      const std::string pool_id{"piscine"};
-      auto pool_deleter = make_scope_guard([&]() {
-          rados.delete_pool(pool_id, boost::asio::detached);
-        });
-      co_await rados.create_pool(pool_id, std::nullopt, 
-          boost::asio::redirect_error(boost::asio::use_awaitable, ec));
-      if (ec) throw boost::system::system_error(ec);
-
-      pool = co_await rados.lookup_pool(pool_id, 
+      co_await rados.create_pool(pool, std::nullopt, 
           boost::asio::redirect_error(boost::asio::use_awaitable, ec));
       if (ec) throw boost::system::system_error(ec);
 
-      co_await rados.delete_pool(pool_id, 
+      pool_id = co_await rados.lookup_pool(pool, 
           boost::asio::redirect_error(boost::asio::use_awaitable, ec));
       if (ec) throw boost::system::system_error(ec);
-      
     }, [](std::exception_ptr e) {
       if (e) std::rethrow_exception(e);
     }
   );
 
-  ASSERT_NO_THROW(ctx.run());
-  ASSERT_GT(pool, 0);
+  EXPECT_NO_THROW(ctx.run());
+  EXPECT_GT(pool_id, 0);
+
+  // cleanup pool
```

Comment:
You could also take an approach like that used in CORO_TEST (and in fact you could make your own version that inherits from it) if you wanted.